checksum_test.go

```diff
@@ -0,0 +1,73 @@
+package getter
+
+import (
+	"encoding/hex"
+	"hash"
+	"net/url"
+	"reflect"
+	"testing"
+)
+
+func u(t *testing.T, in string) *url.URL {
+	u, err := url.Parse(in)
+	if err != nil {
+		t.Fatalf("cannot parse %s: %v", in, err)
+	}
+	return u
+}
+
+func Test_checksumHashAndValue(t *testing.T) {
```

Comment:
Just thinking - I think we should have a full matrix of tests here:

* BSD MD5/SHA
* GNU MD5/SHA

I think in the case of GNU specifically, and both for SHAs, the hash length is also reflective of the algorithm in use, correct? I haven't looked 100%.